### PR TITLE
Fix issue 154

### PR DESCRIPTION
**Pull request by saledouble**
Related to #154

### CODE REVIEW

**Comment by khatchad** [comment]
What about DOI_boundaries.csv?

**Comment by saledouble** [comment]
>What about repos.csv?

Two more columns as shown above.

**Comment by saledouble** [comment]
boundary:


sequence | subject | low boundary inclusive | high boundary exclusive | log level
-- | -- | -- | -- | --
1555430717707 | deviation-upload | 0 | 1.662143 | FINEST
1555430717707 | deviation-upload | 1.662143 | 3.324286 | FINER
1555430717707 | deviation-upload | 3.324286 | 4.986429 | FINE
1555430717707 | deviation-upload | 4.986429 | 6.648572 | CONFIG
1555430717707 | deviation-upload | 6.648572 | 8.310716 | INFO
1555430717707 | deviation-upload | 8.310716 | 9.972858 | WARNING
1555430717707 | deviation-upload | 9.972858 | 11.635 | SEVERE
1555430728038 | deviation-upload | 0 | 1.662143 | FINEST
1555430728038 | deviation-upload | 1.662143 | 3.324286 | FINER
1555430728038 | deviation-upload | 3.324286 | 4.986429 | FINE
1555430728038 | deviation-upload | 4.986429 | 6.648572 | CONFIG
1555430728038 | deviation-upload | 6.648572 | 8.310716 | INFO
1555430728038 | deviation-upload | 8.310716 | 9.972858 | WARNING
1555430728038 | deviation-upload | 9.972858 | 11.635 | SEVERE
1555430739351 | deviation-upload | 0 | 1.939167 | FINEST
1555430739351 | deviation-upload | 1.939167 | 3.878334 | FINER
1555430739351 | deviation-upload | 3.878334 | 5.817501 | FINE
1555430739351 | deviation-upload | 5.817501 | 7.756668 | INFO
1555430739351 | deviation-upload | 7.756668 | 9.695835 | WARNING
1555430739351 | deviation-upload | 9.695835 | 11.635 | SEVERE
1555430752018 | deviation-upload | 0 | 1.939167 | FINEST
1555430752018 | deviation-upload | 1.939167 | 3.878334 | FINER
1555430752018 | deviation-upload | 3.878334 | 5.817501 | FINE
1555430752018 | deviation-upload | 5.817501 | 7.756668 | INFO
1555430752018 | deviation-upload | 7.756668 | 9.695835 | WARNING
1555430752018 | deviation-upload | 9.695835 | 11.635 | SEVERE
1555430762227 | deviation-upload | 0 | 1.939167 | FINEST
1555430762227 | deviation-upload | 1.939167 | 3.878334 | FINER
1555430762227 | deviation-upload | 3.878334 | 5.817501 | FINE
1555430762227 | deviation-upload | 5.817501 | 7.756668 | INFO
1555430772576 | deviation-upload | 0 | 1.939167 | FINEST
1555430772576 | deviation-upload | 1.939167 | 3.878334 | FINER
1555430772576 | deviation-upload | 3.878334 | 5.817501 | FINE
1555430772576 | deviation-upload | 5.817501 | 7.756668 | INFO

input log invocations:


subject | log expression | start pos | log level | type FQN | enclosing method | not lower log levels of logs inside of catch   blocks | log level not transformed due to if condition | DOI value
-- | -- | -- | -- | -- | -- | -- | -- | --
deviation-upload | LOG.info(String.format("%d: %d (%d)",sector_address,sector ==   null ? -1 : sector.end(),address + data.length)) | 16063 | INFO | deviation.Dfu | sendToDevice(deviation.DfuDevice,long,byte[],deviation.Progress) | FALSE | FALSE | 7.056999
deviation-upload | LOG.warning(String.format("Error: Did not find matching device for   VID:0x%x PID:0x%x alt:%d",vid,pid,alt)) | 5235 | WARNING | deviation.DeviationUploader | readBinFromDevice(deviation.DfuDevice,java.lang.String,int,java.lang.Integer,java.lang.Integer,java.lang.Integer,java.lang.Integer,java.lang.Integer,boolean) | FALSE | FALSE | 0
deviation-upload | LOG.severe("Error: Unable to open device") | 5395 | SEVERE | deviation.filesystem.TxInterfaceUSB | open() | FALSE | FALSE | 0.19
deviation-upload | LOG.warning("Error: Unable to open device") | 2647 | WARNING | deviation.DeviationUploader | sendDfuToDevice(deviation.DfuDevice,deviation.DfuFile,deviation.Progress) | FALSE | FALSE | 0.692001
deviation-upload | LOG.warning("Error: Unable to open device") | 5538 | WARNING | deviation.DeviationUploader | readBinFromDevice(deviation.DfuDevice,java.lang.String,int,java.lang.Integer,java.lang.Integer,java.lang.Integer,java.lang.Integer,java.lang.Integer,boolean) | FALSE | FALSE | 0
deviation-upload | LOG.severe(String.format("Cancelled at address   0x%x",sector_address)) | 19412 | SEVERE | deviation.Dfu | sendToDevice(deviation.DfuDevice,long,byte[],deviation.Progress) | FALSE | FALSE | 7.056999
deviation-upload | LOG.info("Starting device read") | 14751 | INFO | deviation.Dfu | fetchFromDevice(deviation.DfuDevice,long,int) | FALSE | FALSE | 0
deviation-upload | LOG.finest(String.format("Sector start: 0x%x end: 0x%x size:%d   count:%d %s%s%s",address,address + size * count - 1,size,count,readable   ? "r" : "",erasable ? "e" :   "",writable ? "w" : "")) | 2193 | FINEST | deviation.DfuMemory$SegmentParser | SegmentParser(java.lang.String,java.lang.String) | FALSE | TRUE | 0
deviation-upload | LOG.finest("Firing Root") | 853 | FINEST | deviation.gui.treemodel.FilesToSendModel | refresh() | FALSE | FALSE | 0.512999
deviation-upload | LOG.finest(String.format("Crc: 0x%x",crc)) | 3677 | FINEST | deviation.Transmitter | calcCrc(long[],int,long,long,long) | FALSE | FALSE | 0
deviation-upload | LOG.info("Could not identify any memory region for rootFS") | 1773 | INFO | deviation.filesystem.TxInterfaceUSB | TxInterfaceUSB(deviation.DfuDevice) | FALSE | FALSE | 0.207001
deviation-upload | LOG.info("Successfully reset STM32") | 21104 | INFO | deviation.Dfu | resetSTM32(deviation.DfuDevice) | FALSE | FALSE | 2.038
deviation-upload | LOG.fine("aborting previous incomplete transfer") | 13307 | FINE | deviation.Dfu | setIdle(deviation.DfuDevice) | FALSE | FALSE | 4.413
deviation-upload | LOG.warning(String.format("DFU Status:   '%s'",status.statusToString())) | 13811 | WARNING | deviation.Dfu | setIdle(deviation.DfuDevice) | FALSE | FALSE | 4.413
deviation-upload | LOG.fine("dfuIDLE, continuing") | 13618 | FINE | deviation.Dfu | setIdle(deviation.DfuDevice) | FALSE | FALSE | 4.413
deviation-upload | LOG.info(String.format(("Cache of 0x%08x :   %d"),range.start(),cached[sector_num] ? 1 : 0)) | 4277 | INFO | deviation.filesystem.FlashIO | cache(int) | FALSE | FALSE | 1.437
deviation-upload | LOG.info(String.format("Count: %d",files.size())) | 6182 | INFO | deviation.gui.FileMgrTab | doInBackground() | FALSE | FALSE | 0
deviation-upload | LOG.severe(String.format("Error: Unable to initiate zero-length   download")) | 20784 | SEVERE | deviation.Dfu | resetSTM32(deviation.DfuDevice) | FALSE | FALSE | 2.038
deviation-upload | LOG.severe(String.format("Error: Non-supported special command   %d",command)) | 4833 | SEVERE | deviation.Dfu | dfuseSpecialCommand(deviation.DfuDevice,long,int) | FALSE | FALSE | 6.017
deviation-upload | LOG.finest("Double click on: " + fileinfo.name()) | 11227 | FINEST | deviation.gui.FileMgrTab$TreeMouseListener | mouseClicked(java.awt.event.MouseEvent) | FALSE | FALSE | 1.059999
deviation-upload | LOG.severe("Device still in Runtime Mode!") | 12793 | SEVERE | deviation.Dfu | setIdle(deviation.DfuDevice) | FALSE | FALSE | 4.413
deviation-upload | LOG.warning("Error: Unable to open device") | 7217 | WARNING | deviation.DeviationUploader | resetDevices(java.util.List) | FALSE | FALSE | 0
deviation-upload | LOG.info(String.format("%s",dev.getTxInfo().type().getName())) | 6602 | INFO | deviation.DeviationUploader | listDevices(java.util.List) | FALSE | FALSE | 0
deviation-upload | LOG.severe(String.format("Error: Page at 0x%x can not be   erased",address)) | 4219 | SEVERE | deviation.Dfu | dfuseSpecialCommand(deviation.DfuDevice,long,int) | FALSE | FALSE | 6.017
deviation-upload | LOG.severe(String.format("Error: xfer_size %d is not a multiple of   the sector size: %d",xfer_size,sector.size())) | 20099 | SEVERE | deviation.Dfu | sendToDevice(deviation.DfuDevice,long,byte[],deviation.Progress) | FALSE | FALSE | 7.056999
deviation-upload | LOG.fine(String.format("DIR: %s",entry.getName())) | 876 | FINE | deviation.filesystem.TxInterfaceCommon | readDirRecur(java.lang.String,de.waldheinz.fs.FsDirectory) | FALSE | FALSE | 1.505
deviation-upload | LOG.warning("Error: Unable to open device") | 4473 | WARNING | deviation.DeviationUploader | sendBinToDevice(deviation.DfuDevice,java.lang.String,int,java.lang.Integer,java.lang.Integer,java.lang.Integer,java.lang.Integer,boolean) | FALSE | FALSE | 0.111
deviation-upload | LOG.severe("Error sending dfu abort request") | 6420 | SEVERE | deviation.Dfu | dfuseSpecialCommand(deviation.DfuDevice,long,int) | FALSE | FALSE | 6.017
deviation-upload | LOG.info(model) | 736 | INFO | deviation.TxInfo | TxInfo(byte[]) | FALSE | FALSE | 0
deviation-upload | LOG.severe(String.format("Error: %d",status.bStatus)) | 14069 | SEVERE | deviation.Dfu | setIdle(deviation.DfuDevice) | FALSE | FALSE | 4.413
deviation-upload | LOG.info("Resetting STM32, starting firmware at address   0x08000000...") | 20416 | INFO | deviation.Dfu | resetSTM32(deviation.DfuDevice) | FALSE | FALSE | 2.038
deviation-upload | LOG.severe("Error: Failed to enter idle state on abort") | 6638 | SEVERE | deviation.Dfu | dfuseSpecialCommand(deviation.DfuDevice,long,int) | FALSE | FALSE | 6.017
deviation-upload | LOG.fine("dfuERROR, clearing status") | 12940 | FINE | deviation.Dfu | setIdle(deviation.DfuDevice) | FALSE | FALSE | 4.413
deviation-upload | LOG.info(String.format("0x%08x Read: %s Write:   %s",range.start(),chksum[sector_num],newchksum)) | 2592 | INFO | deviation.filesystem.FlashIO | close() | FALSE | FALSE | 0.003
deviation-upload | LOG.info(String.format("Device\t%9s %8s   %8s %7s     %s","Interface","Start","End","Size","Count")) | 6427 | INFO | deviation.DeviationUploader | listDevices(java.util.List) | FALSE | FALSE | 0
deviation-upload | LOG.info("Hotplug detected") | 2249 | INFO | deviation.gui.MonitorUSB | doInBackground() | FALSE | FALSE | 1.162001
deviation-upload | LOG.info("flush") | 3222 | INFO | deviation.filesystem.FlashIO | flush() | FALSE | FALSE | 0.02
deviation-upload | LOG.info(String.format("Setting address pointer to   0x%x",address)) | 4670 | INFO | deviation.Dfu | dfuseSpecialCommand(deviation.DfuDevice,long,int) | FALSE | FALSE | 6.017
deviation-upload | LOG.warning(String.format("Only read %d bytes from %s:%s (expected   %d)",ostream.size(),zipFile,fname,ze.getSize())) | 1418 | WARNING | deviation.ZipFile | GetFileList(java.lang.String) | FALSE | FALSE | 0
deviation-upload | LOG.warning("Failed to find start sector") | 3432 | WARNING | deviation.filesystem.DevoFS.DevoFSLayout | removeSectorBoundaries(java.nio.ByteBuffer) | FALSE | FALSE | 0
deviation-upload | LOG.severe(String.format("Erasing page size %d at address 0x%x, page   " + "starting at 0x%x",sector.size(),address,address &   ~(sector.size() - 1))) | 4352 | SEVERE | deviation.Dfu | dfuseSpecialCommand(deviation.DfuDevice,long,int) | FALSE | FALSE | 6.017
deviation-upload | LOG.severe("can't send DFU_ABORT") | 13426 | SEVERE | deviation.Dfu | setIdle(deviation.DfuDevice) | FALSE | FALSE | 4.413
deviation-upload | LOG.fine(String.format("Poll timeout %d   ms",status.bwPollTimeout)) | 5592 | FINE | deviation.Dfu | dfuseSpecialCommand(deviation.DfuDevice,long,int) | FALSE | FALSE | 6.017
deviation-upload | LOG.info(entry.getName()) | 1754 | INFO | deviation.filesystem.TxInterfaceCommon | readDir(de.waldheinz.fs.FileSystem,java.lang.String) | FALSE | FALSE | 0.088
deviation-upload | LOG.severe("Error during special command download") | 5220 | SEVERE | deviation.Dfu | dfuseSpecialCommand(deviation.DfuDevice,long,int) | FALSE | FALSE | 6.017
deviation-upload | LOG.severe(String.format("Error: Write failed to set address:   0x%x",sector_address)) | 16751 | SEVERE | deviation.Dfu | sendToDevice(deviation.DfuDevice,long,byte[],deviation.Progress) | FALSE | FALSE | 7.056999
deviation-upload | LOG.severe("error clear_status") | 13053 | SEVERE | deviation.Dfu | setIdle(deviation.DfuDevice) | FALSE | FALSE | 4.413
deviation-upload | LOG.severe("Error: Wrong state after command download") | 5448 | SEVERE | deviation.Dfu | dfuseSpecialCommand(deviation.DfuDevice,long,int) | FALSE | FALSE | 6.017
deviation-upload | LOG.severe("Error: Unable to open media device") | 7038 | SEVERE | deviation.filesystem.TxInterfaceUSB | getFSStatus() | TRUE | FALSE | 0.924
deviation-upload | LOG.severe(String.format("Error: No sector found that can be written   at address 0x%08x",sector_address)) | 16253 | SEVERE | deviation.Dfu | sendToDevice(deviation.DfuDevice,long,byte[],deviation.Progress) | FALSE | FALSE | 7.056999
deviation-upload | LOG.info(String.format("Bytes/sec: %.6f Avg: %.6f",1000.0 *   range.size() / (endTime - startTime),1000.0 * totalBytes / totalTime)) | 4719 | INFO | deviation.filesystem.FlashIO | cache(int) | FALSE | FALSE | 1.437
deviation-upload | LOG.severe("Error: Invalid descriptor list") | 778 | SEVERE | deviation.DfuFuncDescriptor | find_descriptor(byte[],int,int) | FALSE | FALSE | 0
deviation-upload | LOG.info(String.format("dfuseUpload: libusb_control_transfer   returned %d",bytes_received)) | 7689 | INFO | deviation.Dfu | dfuseUpload(deviation.DfuDevice,byte[],int) | FALSE | FALSE | 0
deviation-upload | LOG.warning(String.format("Transmitter ID: '%s'",new   String(info.getIdentifier()))) | 3285 | WARNING | deviation.DeviationUploader | sendDfuToDevice(deviation.DfuDevice,deviation.DfuFile,deviation.Progress) | FALSE | FALSE | 0.692001
deviation-upload | LOG.warning("Error: Unable to open device") | 2674 | WARNING | deviation.TxInfo | getTxInfo(deviation.DfuDevice) | FALSE | FALSE | 0
deviation-upload | LOG.severe(String.format("Error: Write failed to set address:   0x%x",sector_address)) | 17989 | SEVERE | deviation.Dfu | sendToDevice(deviation.DfuDevice,long,byte[],deviation.Progress) | FALSE | FALSE | 7.056999
deviation-upload | LOG.severe("Error: Unable to open root device") | 7255 | SEVERE | deviation.filesystem.TxInterfaceUSB | getFSStatus() | TRUE | FALSE | 0.924
deviation-upload | LOG.warning(String.format("Error: Dfu Tx type '%s' does not match   transmitter type   '%s'",TxInfo.typeToString(TxInfo.getModelFromString(elem.name())),TxInfo.typeToString(info.type()))) | 2935 | WARNING | deviation.DeviationUploader | sendDfuToDevice(deviation.DfuDevice,deviation.DfuFile,deviation.Progress) | FALSE | FALSE | 0.692001
deviation-upload | LOG.warning(String.format("Only read %d bytes from %s:%s (expected   %d)",ostream.size(),zipFile,fname,ze.getSize())) | 2655 | WARNING | deviation.ZipFile | read(java.lang.String) | FALSE | FALSE | 0
deviation-upload | LOG.info(String.format("\t\t%d %08x     %08x %7d     %d",i,sector.start(),sector.end(),sector.size(),sector.count())) | 6866 | INFO | deviation.DeviationUploader | listDevices(java.util.List) | FALSE | FALSE | 0
deviation-upload | LOG.info("No devices found.") | 6371 | INFO | deviation.DeviationUploader | listDevices(java.util.List) | FALSE | FALSE | 0
deviation-upload | LOG.info(String.format("dfuseDownload: libusb_control_transfer   returned %d",bytes_sent)) | 8629 | INFO | deviation.Dfu | dfuseDownload(deviation.DfuDevice,byte[],int) | FALSE | FALSE | 0
deviation-upload | LOG.info(String.format("Found device: %s [%05x:%04x] cfg=%d,   intf=%d, alt=%d, name='%s'",dev.DFU_IFF_DFU() ? "DFU" :   "Runtime",dev.idVendor(),dev.idProduct(),dev.bConfigurationValue(),dev.bInterfaceNumber(),dev.bAlternateSetting(),mem   == null ? dev.GetId() : mem.name())) | 8914 | INFO | deviation.DeviationUploader | main(java.lang.String[]) | FALSE | FALSE | 11.635
deviation-upload | LOG.fine(TEST_FAT_FILE + " file not found. Continue without.") | 1161 | FINE | deviation.filesystem.TxInterfaceEmulator | TxInterfaceEmulator() | TRUE | FALSE | 0.855999
deviation-upload | LOG.fine(String.format("FILE: %s   (%d)",entry.getName(),entry.getFile().getLength())) | 1134 | FINE | deviation.filesystem.TxInterfaceCommon | readDirRecur(java.lang.String,de.waldheinz.fs.FsDirectory) | FALSE | FALSE | 1.505
deviation-upload | LOG.fine(String.format("bytes_per_hash=%d",xfer_size)) | 14685 | FINE | deviation.Dfu | fetchFromDevice(deviation.DfuDevice,long,int) | FALSE | FALSE | 0
deviation-upload | LOG.info("Encrypting txid for " +   TxInfo.typeToString(info.type())) | 1042 | INFO | deviation.DeviationUploader | applyEncryption(deviation.DfuFile.ImageElement,deviation.TxInfo) | FALSE | FALSE | 0
deviation-upload | LOG.info("FILE: " + file.name()) | 6276 | INFO | deviation.gui.FileMgrTab | doInBackground() | FALSE | FALSE | 0
deviation-upload | LOG.info(String.format("Erasing page: 0x%x",sector_address)) | 17560 | INFO | deviation.Dfu | sendToDevice(deviation.DfuDevice,long,byte[],deviation.Progress) | FALSE | FALSE | 7.056999
deviation-upload | LOG.warning(String.format("Unexpected string: %s",str)) | 3191 | WARNING | deviation.DfuMemory | DfuMemory(java.lang.String) | FALSE | FALSE | 0
deviation-upload | LOG.severe(String.format("Error: state(%d) = %s, status(%d) =   %s",status.bState,status.stateToString(),status.bStatus,status.statusToString())) | 9645 | SEVERE | deviation.Dfu | dfuseDownloadChunk(deviation.DfuDevice,byte[],int) | FALSE | FALSE | 0.365997
deviation-upload | LOG.severe("Failed to create file '" + file.name()) | 2330 | SEVERE | deviation.filesystem.FSUtils | copyFile(de.waldheinz.fs.FileSystem,deviation.FileInfo) | FALSE | FALSE | 0.566999
deviation-upload | LOG.warning(String.format("Error: Did not find matching device for   VID:0x%x PID:0x%x alt:%d",vid,pid,alt)) | 4262 | WARNING | deviation.DeviationUploader | sendBinToDevice(deviation.DfuDevice,java.lang.String,int,java.lang.Integer,java.lang.Integer,java.lang.Integer,java.lang.Integer,boolean) | FALSE | FALSE | 0.111
deviation-upload | LOG.severe(String.format("Error during second get_status" +   " -- state(%d) = %s, status(%d) =   %s",status.bState,status.stateToString(),status.bStatus,status.statusToString())) | 5935 | SEVERE | deviation.Dfu | dfuseSpecialCommand(deviation.DfuDevice,long,int) | FALSE | FALSE | 6.017
deviation-upload | LOG.severe("Error: Expected STM32 to be in dfuMANIFEST state after   get-status command!") | 20988 | SEVERE | deviation.Dfu | resetSTM32(deviation.DfuDevice) | FALSE | FALSE | 2.038
deviation-upload | LOG.info("Transitioning to dfuMANIFEST state") | 9509 | INFO | deviation.Dfu | dfuseDownloadChunk(deviation.DfuDevice,byte[],int) | FALSE | FALSE | 0.365997
deviation-upload | LOG.severe(String.format("Error: Dfu Tx type '%s' does not match   transmitter type   '%s'",TxInfo.typeToString(type),TxInfo.typeToString(gui.getTxInfo().type()))) | 6147 | SEVERE | deviation.gui.DfuSendTab$FileChooserBtnListener | actionPerformed(java.awt.event.ActionEvent) | FALSE | FALSE | 0.991999
deviation-upload | LOG.warning(String.format("Error: Did not find matching device for   VID:0x%x PID:0x%x   alt:%d",file.idVendor(),file.idProduct(),elem.altSetting())) | 2381 | WARNING | deviation.DeviationUploader | sendDfuToDevice(deviation.DfuDevice,deviation.DfuFile,deviation.Progress) | FALSE | FALSE | 0.692001
deviation-upload | LOG.warning("Couldn't get root dir: " + e.getMessage()) | 882 | WARNING | deviation.filesystem.FSUtils | getFileDirectory(de.waldheinz.fs.FileSystem,deviation.FileInfo,boolean) | TRUE | FALSE | 2.001
deviation-upload | LOG.severe(String.format("Error: Write failed to erase address:   0x%x",sector_address)) | 17732 | SEVERE | deviation.Dfu | sendToDevice(deviation.DfuDevice,long,byte[],deviation.Progress) | FALSE | FALSE | 7.056999
deviation-upload | LOG.severe(file.name() + " exists but is not a file") | 2446 | SEVERE | deviation.filesystem.FSUtils | copyFile(de.waldheinz.fs.FileSystem,deviation.FileInfo) | FALSE | FALSE | 0.566999
deviation-upload | LOG.fine(String.format("Determining device status: state=%s   status=%d",status.stateToString(),status.bStatus)) | 12493 | FINE | deviation.Dfu | setIdle(deviation.DfuDevice) | FALSE | FALSE | 4.413
deviation-upload | LOG.info("Unplug detected") | 1847 | INFO | deviation.gui.MonitorUSB | doInBackground() | FALSE | FALSE | 1.162001
deviation-upload | LOG.info(String.format("Font: %d",f.getSize())) | 2356 | INFO | deviation.gui.TextEditor | TextEditor(deviation.FileInfo) | FALSE | FALSE | 0.495999
deviation-upload | LOG.severe(String.format("Cancelled at address   0x%x",sector_address)) | 17367 | SEVERE | deviation.Dfu | sendToDevice(deviation.DfuDevice,long,byte[],deviation.Progress) | FALSE | FALSE | 7.056999
deviation-upload | LOG.warning("Couldn't create directory '" + subdir) | 1373 | WARNING | deviation.filesystem.FSUtils | getFileDirectory(de.waldheinz.fs.FileSystem,deviation.FileInfo,boolean) | FALSE | FALSE | 2.001
deviation-upload | LOG.severe(String.format("Error: Write failed to write address :   0x%x",sector_address)) | 19713 | SEVERE | deviation.Dfu | sendToDevice(deviation.DfuDevice,long,byte[],deviation.Progress) | FALSE | FALSE | 7.056999
deviation-upload | LOG.finest(String.format("%02x ",extra[i])) | 2487 | FINEST | deviation.DfuFuncDescriptor | DfuFuncDescriptor(deviation.DfuDevice) | FALSE | FALSE | 0.312
deviation-upload | LOG.finest(String.format("%04x:%04x -> %02x/%02x   %02x/%02x",desc.idVendor() & 0xffff,desc.idProduct() &   0xffff,intf.bInterfaceNumber(),intf.bAlternateSetting(),intf.bInterfaceClass(),intf.bInterfaceSubClass())) | 1857 | FINEST | deviation.Dfu | findInterfaces(de.ailis.usb4java.libusb.Device) | FALSE | TRUE | 0
deviation-upload | LOG.severe("Error: Unable to set start address for resetting") | 20603 | SEVERE | deviation.Dfu | resetSTM32(deviation.DfuDevice) | FALSE | FALSE | 2.038
deviation-upload | LOG.severe(String.format("Error: Write failed to set address:   0x%x",sector_address)) | 18818 | SEVERE | deviation.Dfu | sendToDevice(deviation.DfuDevice,long,byte[],deviation.Progress) | FALSE | FALSE | 7.056999

repos:

sequence | repo URL | SHA-1 of head | N for commits | number of commits processed | actual number of commits | average Java lines added | average Java lines removed
-- | -- | -- | -- | -- | -- | -- | --
1555430717707 | https://github.com/ponder-lab/deviation-upload.git | ababe489f0a0532cc5373377ca6e260afe4cb4a9 | 1000 | 79 | 79 | 238 | 75
1555430728038 | https://github.com/ponder-lab/deviation-upload.git | ababe489f0a0532cc5373377ca6e260afe4cb4a9 | 1000 | 79 | 79 | 238 | 75
1555430739351 | https://github.com/ponder-lab/deviation-upload.git | ababe489f0a0532cc5373377ca6e260afe4cb4a9 | 1000 | 79 | 79 | 238 | 75
1555430752018 | https://github.com/ponder-lab/deviation-upload.git | ababe489f0a0532cc5373377ca6e260afe4cb4a9 | 1000 | 79 | 79 | 238 | 75
1555430762227 | https://github.com/ponder-lab/deviation-upload.git | ababe489f0a0532cc5373377ca6e260afe4cb4a9 | 1000 | 79 | 79 | 238 | 75
1555430772576 | https://github.com/ponder-lab/deviation-upload.git | ababe489f0a0532cc5373377ca6e260afe4cb4a9 | 1000 | 79 | 79 | 238 | 75


results

sequence | subject | repo URL | input logging statements | passing logging statements | failures | transformed logging statements | log level not lowered in a catch block | log level not transformed due to if condition | use log category (SEVERE/WARNING/CONFIG) | use log category (CONFIG) | not lower log levels of logs inside of catch   blocks | time (s)
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
1555430717707 | deviation-upload | https://github.com/ponder-lab/deviation-upload.git | 91 | 91 | 0 | 79 | 0 | 2 | FALSE | FALSE | FALSE | 10.003
1555430728038 | deviation-upload | https://github.com/ponder-lab/deviation-upload.git | 91 | 91 | 0 | 75 | 4 | 2 | FALSE | FALSE | TRUE | 10.718
1555430739351 | deviation-upload | https://github.com/ponder-lab/deviation-upload.git | 91 | 91 | 0 | 78 | 0 | 2 | FALSE | TRUE | FALSE | 12.403
1555430752018 | deviation-upload | https://github.com/ponder-lab/deviation-upload.git | 91 | 91 | 0 | 74 | 4 | 2 | FALSE | TRUE | TRUE | 10.001
1555430762227 | deviation-upload | https://github.com/ponder-lab/deviation-upload.git | 91 | 91 | 0 | 28 | 0 | 2 | TRUE | FALSE | FALSE | 10.307
1555430772576 | deviation-upload | https://github.com/ponder-lab/deviation-upload.git | 91 | 91 | 0 | 27 | 4 | 2 | TRUE | FALSE | TRUE | 9.752

File: edu.cuny.hunter.log.evalution/src/edu/cuny/hunter/log/evaluation/handlers/EvaluationHandler.java [hunk 1]
```diff
@@ -135,19 +141,22 @@ public Object execute(ExecutionEvent event) throws ExecutionException {
...
+										"Fatal error encountered during evaluation: "
+												+ status.getMessageMatchingSeverity(RefactoringStatus.FATAL));
 
 							resultsTimeCollector.stop();
 
 							Set<LogInvocation> logInvocationSet = logRejuvenatingProcessor.getLogInvocationSet();
 
 							// Just print once.
-							if (i == 0)
+							if (i == 1)
```
Comment (khatchad) [inline edu.cuny.hunter.log.evalution/src/edu/cuny/hunter/log/evaluation/handlers/EvaluationHandler.java]: What is the purpose of this change?

Comment (saledouble) [reply edu.cuny.hunter.log.evalution/src/edu/cuny/hunter/log/evaluation/handlers/EvaluationHandler.java]: For the first configuration, the setting for the catch block is turned off. The result will be 0 because of that.

**Comment by saledouble** [comment]
[results.zip](https://github.com/ponder-lab/Logging-Level-Evolution-Plugin/files/3086168/results.zip)

**Comment by saledouble** [comment]
[results.zip](https://github.com/ponder-lab/Logging-Level-Evolution-Plugin/files/3086493/results.zip)

suffix `154` is after changes
suffix `master` is before changes
suffix `single` is testing `deivation`
suffic `double` is testing `deviation` and `IRCT-API`

**Comment by saledouble** [comment]
Closed. We have new PR: #192
